### PR TITLE
feat: mobile board view

### PR DESCRIPTION
**Pull request by hyj1204**
### Feature Preview
- <details>
  <summary> Improve card view on the board</summary>
  All types of card cell builder widgets are under `appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells` image
</details>

- Long press to move card and group

- Add a new card
   
https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/43774e8f-11d5-4975-9587-2720cfb5cc66

- Add a new group

https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/37906bea-d346-416c-942a-d00540e50731

- Rename, hide and unhide a group

https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/20517b92-7dff-4467-881f-b874ece391c3

---

#### PR Checklist

- [X] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [X] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/mobile/presentation/database/board/mobile_board_content.dart [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:go_router/go_router.dart';
+
+class MobileBoardContent extends StatefulWidget {
+  const MobileBoardContent({
+    super.key,
+    this.onEditStateChanged,
+  });
+
+  final VoidCallback? onEditStateChanged; //**??? what is this for?
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/board/mobile_board_content.dart]: I'm assuming it's related to editing a cards "title" property.

.. Which isn't possible on Mobile.

File: frontend/appflowy_flutter/lib/mobile/presentation/database/board/mobile_board_content.dart [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+          color: style.colorScheme.onSurface,
+        ),
+        label: Text(
+          LocaleKeys.board_column_createNewCard.tr(),
+          style: style.textTheme.bodyMedium?.copyWith(
+            color: style.colorScheme.onSurface,
+          ),
+        ),
+        onPressed: () => context.read<BoardBloc>()
+          ..add(BoardEvent.createBottomRow(columnData.id)),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/board/mobile_board_content.dart:148]:
```suggestion
          .add(BoardEvent.createBottomRow(columnData.id)),
```

File: frontend/appflowy_flutter/lib/mobile/presentation/database/board/mobile_board_content.dart [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+        initialChildSize: 0.4,
+        minChildSize: 0.4,
+        snapSizes: const [0.4, 1],
+        snap: true,
+        builder: (context, scrollController) {
+          return MobileCardDetailScreen(
+            fieldController: fieldController,
+            rowController: dataController,
+            scrollController: scrollController,
+            isBottemSheet: true,
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/board/mobile_board_content.dart:277]:
```suggestion
            isBottomSheet: true,
```

File: frontend/appflowy_flutter/lib/mobile/presentation/database/board/widgets/group_card_header.dart [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        offset: widget.groupData.headerData.groupName.length,
+      ),
+      text: widget.groupData.headerData.groupName,
+    ),
+  );
+
+  @override
+  void initState() {
+    super.initState();
+  }
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/board/widgets/group_card_header.dart:43]: Redundant

File: frontend/appflowy_flutter/lib/mobile/presentation/database/board/widgets/mobile_board_trailing.dart [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  Widget build(BuildContext context) {
+    final screenSize = MediaQuery.of(context).size;
+    final style = Theme.of(context);
+
+    return Container(
+      margin: const EdgeInsets.all(8),
+      child: SizedBox(
+        width: screenSize.width * 0.7,
+        child: isEditing
+            ? Container(
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/board/widgets/mobile_board_trailing.dart:32]:
```suggestion
            ? DecoratedBox(
```

File: frontend/appflowy_flutter/lib/mobile/presentation/database/board/widgets/mobile_hidden_groups_column.dart [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+    final databaseController = context.read<BoardBloc>().databaseController;
+    return BlocSelector<BoardBloc, BoardState, BoardLayoutSettingPB?>(
+      selector: (state) => state.layoutSettings,
+      builder: (context, layoutSettings) {
+        if (layoutSettings == null) {
+          return const SizedBox.shrink();
+        }
+        final isCollapsed = layoutSettings.collapseHiddenGroups;
+        return Container(
+          margin: margin,
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/board/widgets/mobile_hidden_groups_column.dart:37]: Padding?

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/checkbox.dart [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      value: _cellBloc,
+      child: BlocBuilder<CheckboxCellBloc, CheckboxCellState>(
+        buildWhen: (previous, current) =>
+            previous.isSelected != current.isSelected,
+        builder: (context, state) {
+          final icon = FlowySvg(
+            state.isSelected ? FlowySvgs.check_filled_s : FlowySvgs.uncheck_s,
+            blendMode: BlendMode.dst,
+            size: const Size.square(24),
+          );
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/checkbox.dart:44]: redundant variable, only accessed once

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/checklist.dart [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+class _ChecklistCellState extends State<MobileChecklistCardCell> {
+  late ChecklistCellBloc _cellBloc;
+
+  @override
+  void initState() {
+    final cellController =
+        widget.cellControllerBuilder.build() as ChecklistCellController;
+    _cellBloc = ChecklistCellBloc(cellController: cellController);
+    _cellBloc.add(const ChecklistCellEvent.initial());
+    super.initState();
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/checklist.dart:31]: super.initState call should be first

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/checklist.dart [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+class MobileChecklistProgressBar extends StatefulWidget {
+  final List<ChecklistSelectOption> tasks;
+  final double percent;
+  final int segmentLimit = 5;
+
+  const MobileChecklistProgressBar({
+    required this.tasks,
+    required this.percent,
+    Key? key,
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/checklist.dart:65]: Super key constructor

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/date.dart [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+  final CellControllerBuilder cellControllerBuilder;
+  final CellRenderHook<dynamic, CustomCardData>? renderHook;
+
+  @override
+  State<MobileDateCardCell> createState() => _DateCellState();
+}
+
+class _DateCellState extends State<MobileDateCardCell> {
+  late DateCellBloc _cellBloc;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/date.dart:23]: Can be `late final`

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/date.dart [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+  @override
+  void initState() {
+    final cellController =
+        widget.cellControllerBuilder.build() as DateCellController;
+
+    _cellBloc = DateCellBloc(cellController: cellController)
+      ..add(const DateCellEvent.initial());
+    super.initState();
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/date.dart:32]: same here - should be called first

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/number.dart [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+  late NumberCellBloc _cellBloc;
+
+  @override
+  void initState() {
+    final cellController =
+        widget.cellControllerBuilder.build() as NumberCellController;
+
+    _cellBloc = NumberCellBloc(cellController: cellController)
+      ..add(const NumberCellEvent.initial());
+    super.initState();
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/number.dart:33]: Same

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/select_option.dart [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+class _SelectOptionCellState extends State<MobileSelectOptionCardCell> {
+  late SelectOptionCellBloc _cellBloc;
+
+  @override
+  void initState() {
+    final cellController =
+        widget.cellControllerBuilder.build() as SelectOptionCellController;
+    _cellBloc = SelectOptionCellBloc(cellController: cellController)
+      ..add(const SelectOptionCellEvent.initial());
+    super.initState();
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/select_option.dart:36]: Same

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/select_option.dart [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+  final String name;
+  final Color color;
+  final VoidCallback? onSelected;
+  final void Function(String)? onRemove;
+  const MobileSelectOptionTag({
+    required this.name,
+    required this.color,
+    this.onSelected,
+    this.onRemove,
+    Key? key,
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/select_option.dart:97]: super key constructor

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/text.dart [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+  late TextCellBloc _cellBloc;
+
+  @override
+  void initState() {
+    final cellController =
+        widget.cellControllerBuilder.build() as TextCellController;
+
+    _cellBloc = TextCellBloc(cellController: cellController)
+      ..add(const TextCellEvent.initial());
+    super.initState();
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/text.dart:33]: Same

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/time_stamp.dart [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+  late TimestampCellBloc _cellBloc;
+
+  @override
+  void initState() {
+    final cellController =
+        widget.cellControllerBuilder.build() as TimestampCellController;
+
+    _cellBloc = TimestampCellBloc(cellController: cellController)
+      ..add(const TimestampCellEvent.initial());
+    super.initState();
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/time_stamp.dart:32]: Same

File: frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/url.dart [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+class _URLCellState extends State<MobileURLCardCell> {
+  late URLCellBloc _cellBloc;
+
+  @override
+  void initState() {
+    final cellController =
+        widget.cellControllerBuilder.build() as URLCellController;
+    _cellBloc = URLCellBloc(cellController: cellController);
+    _cellBloc.add(const URLCellEvent.initial());
+    super.initState();
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_content/card_cells/url.dart:29]: Same

File: frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/board_page.dart [hunk 18]
```diff
@@ -89,39 +91,49 @@ class BoardPage extends StatelessWidget {
...
-            ),
+            (_) => PlatformExtension.isMobile
+                ? MobileBoardContent(
+                    onEditStateChanged: onEditStateChanged,
+                  )
+                : DesktopBoardContent(onEditStateChanged: onEditStateChanged),
+            (err) => PlatformExtension.isMobile
+                ? FlowyMobileStateContainer.error(
+                    emoji: '🛸',
+                    title: 'Failed to load board view',
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/board/presentation/board_page.dart:98]: Localization

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/card/card_cell_builder.dart [hunk 19]
```diff
@@ -34,6 +36,42 @@ class CardCellBuilder<CustomCardData> {
...
+            cellContext: cellContext,
+            cellControllerBuilder: cellControllerBuilder,
+            style: style,
+            cardData: cardData,
+            cellNotifier: cellNotifier,
+            renderHook: renderHook,
+            hasNotes: hasNotes,
+          );
+
+    return cellWidget;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/card/card_cell_builder.dart:62]:
```suggestion
    return PlatformExtension.isMobile
        ? _getMobileCardCellWidget(
            key: key,
            cellContext: cellContext,
            cellControllerBuilder: cellControllerBuilder,
            style: style,
            cardData: cardData,
            cellNotifier: cellNotifier,
            renderHook: renderHook,
            hasNotes: hasNotes,
          )
        : _getDesktopCardCellWidget(
            key: key,
            cellContext: cellContext,
            cellControllerBuilder: cellControllerBuilder,
            style: style,
            cardData: cardData,
            cellNotifier: cellNotifier,
            renderHook: renderHook,
            hasNotes: hasNotes,
          );
```

**Review by Xazin — APPROVED** [review]

File: frontend/appflowy_flutter/lib/mobile/presentation/base/mobile_view_page.dart [hunk 20]
```diff
@@ -114,33 +114,39 @@ class _MobileViewPageState extends State<MobileViewPage> {
...
-                fontSize: 22.0,
-              ),
-            Expanded(
-              child: FlowyText.regular(
-                view?.name ?? widget.title ?? '',
-                fontSize: 14.0,
-                overflow: TextOverflow.ellipsis,
+    return WillPopScope(
+      // prevent navigate to previous screen by scrolling left on the screen
+      onWillPop: () async {
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/mobile/presentation/base/mobile_view_page.dart:119]: It's definitely not a good solution to disable the swipe-left gesture.

You should add padding between the swipe-left gesture hit test area and the scrollable widget area instead.

**Pull request merged**